src/devices/Scd4x/Scd4x.cs

```diff
@@ -0,0 +1,240 @@
+        ///
+        /// <para>
+        /// Every period, the length of which is available in <see cref="MeasurementPeriod"/>, <see cref="CheckDataReady"/> can be called to see if a measurement is available,
+        /// and the measurement can then be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StopPeriodicMeasurements"/> must be called to stop periodic measurements.
+        /// </para>
+        /// </summary>
+        public void StartPeriodicMeasurements() =>
+            _device.Write(StartPeriodicMeasurementBytes);
+
+        /// <summary>
+        /// <para>
+        /// Checks if a periodic measurement is available.
+        /// Once available, the measurement can be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StartPeriodicMeasurements"/> must be called first.
+        /// </para>
+        /// </summary>
+        /// <returns>If a measurement is available, <see langword="true"/>. Otherwise, <see langword="false"/>.</returns>
+        public bool CheckDataReady()
+        {
+            TimeSpan delay = BeginCheckDataReady();
+            Thread.Sleep(delay);
+            return EndCheckDataReady();
+        }
+
+        /// <summary>
+        /// <para>
+        /// Begins checking if a periodic measurement is available.
+        /// Once available, the measurement can be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StartPeriodicMeasurements"/> must be called first.
+        /// </para>
+        /// </summary>
+        /// <returns>
+        /// The time that the caller must wait before calling <see cref="EndCheckDataReady"/> to retreive the current data ready status.
+        /// </returns>
+        public TimeSpan BeginCheckDataReady()
+        {
+            _device.Write(CheckDataReadyStatusBytes);
+            return TimeSpan.FromTicks(TimeSpan.TicksPerMillisecond);
+        }
+
+        /// <summary>
+        /// Completes checking if a periodic measurement is available.
+        /// <see cref="BeginCheckDataReady"/> must have been called first.
+        /// </summary>
+        /// <returns>If a measurement is available, <see langword="true"/>. Otherwise, <see langword="false"/>.</returns>
+        public bool EndCheckDataReady()
+        {
+            Span<byte> buffer = stackalloc byte[3];
+            _device.Read(buffer);
+
+            return Sensirion.ReadUInt16BigEndianAndCRC8(buffer) is ushort response && (response & 0x7FF) != 0;
+        }
+
+        /// <summary>
+        /// <para>
+        /// Reads a periodic CO₂, humidity, and temperature measurement from the sensor.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="CheckDataReady"/> should be called first to ensure a measurement is available. Once read, data will no longer be ready.
```

Comment:
Is there any reason why this method's implementation shouldn't call CheckDataReady? If there is not, you could move that here and then get rid of a lot of public surface area like the begin/end data ready as well as the CheckDataReady, and only leave the ReadPeriodicMeasurment member public

Reply:
Good point, I've made this change.